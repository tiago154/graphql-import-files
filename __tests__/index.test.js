const { loadFile, loadFiles } = require('../src/index')

describe('loadFile', () => {
  describe('success', () => {
    it('should valid schema', () => {
      const schema = loadFile('./__tests__/mocks/file/user.test.graphql')

      expect(schema).toMatchSnapshot()
    })
  })

  describe('failure', () => {
    it('should an error when not finding the file', () => {
      try {
        loadFile('./__tests__/mocks/file/error.graphql')
      } catch (error) {
        expect(error.message).toContain('__tests__\\mocks\\file\\error.graphql')
        expect(error.code).toBe('ENOENT')
      }
    })
  })
})

describe('loadFiles', () => {
  describe('success', () => {
    it('should valid schema with many file', () => {
      const gropPattern = '**/__tests__/mocks/multifiles/**/*.{graphql,gql}'
      const schema = loadFiles(gropPattern)

      expect(schema).toMatchSnapshot()
    })
  })

  describe('failure', () => {
    it('should returns an empty array when no file is founds', () => {
      const gropPattern = '**/__tests__/mocks/multifiles/**/*.{error,fail}'
      const schema = loadFiles(gropPattern)

      expect(schema).toEqual([])
    })
  })
})