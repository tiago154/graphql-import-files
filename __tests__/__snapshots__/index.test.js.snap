// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadFile success should valid schema 1`] = `
"type UserTest {
  id: Int!
  name: String!
  age: Int
  tags: [String!]!
}"
`;

exports[`loadFiles success should valid schema with many file 1`] = `
Array [
  "type Product {
  id: ID!
  name: String!
  price: Float!
}",
  "type Query {
  hello: String!
}",
  "type User {
  id: Int!
  name: String
}",
]
`;
